src/background/starterBlueprints.ts

```diff
@@ -23,74 +23,147 @@ import { forEachTab } from "@/background/util";
 import { queueReactivateTab } from "@/contentScript/messenger/api";
 import { ExtensionOptionsState } from "@/store/extensionsTypes";
 import reportError from "@/telemetry/reportError";
+import { debounce } from "lodash";
 
 const { reducer, actions } = extensionsSlice;
 
-function installStarterBlueprint(
+const PLAYGROUND_URL = "https://www.pixiebrix.com/playground";
+let isInstallingBlueprints = false;
+const BLUEPRINT_INSTALLATION_DEBOUNCE_MS = 10_000;
+const BLUEPRINT_INSTALLATION_MAX_MS = 60_000;
+
+function installBlueprint(
   state: ExtensionOptionsState,
-  starterBlueprint: RecipeDefinition
+  blueprint: RecipeDefinition
 ): ExtensionOptionsState {
   return reducer(
     state,
     actions.installRecipe({
-      recipe: starterBlueprint,
-      extensionPoints: starterBlueprint.extensionPoints,
+      recipe: blueprint,
+      extensionPoints: blueprint.extensionPoints,
     })
   );
 }
 
-export async function installStarterBlueprints(): Promise<void> {
+async function installBlueprints(
```

Comment:
To ensure that it's not called _undebounced_ (as it currently is on line 150), we can prepend `_` to the name. I generally inline the function in the `debounce()` call but for long once it's less readable.

```suggestion
async function _installBlueprints(
```